Sources/NetworkService/NetworkService.swift

```diff
@@ -1,8 +1,82 @@
-/// A stub object for future use as a network layer.
-final public class NetworkService {
-    /// Initialize the new instance of `NetworkService`
-    public init() {}
-    
-    /// A test method that does nothing. You can use the current method to check the availability of the API in your module.
-    public func testUse() {}
+import Foundation
+
+/// Handles sending requests and receiving responses.
+public final class NetworkService<Endpoint: HTTPAPIEndpoint>: NetworkRouter {
+    /// The default timeout interval (in seconds) for network requests.
+    public var defaultTimeoutIntervalInSeconds: TimeInterval = 10.0
+
+    /// Sends a network request to the specified server endpoint.
+    ///
+    /// - Parameter route: The endpoint representing the target server location and details of the request.
+    /// - Returns: A `Result` object containing either the received data or an error in case of failure.
+    public func request(_ route: Endpoint) async -> Result<Data, Error> {
+        return await send(request(from: route))
+    }
+
+    // MARK: - Private interface
+
+    private var task: URLSessionTask?
+
+    private func request(from route: Endpoint) -> URLRequest {
+        var request = URLRequest(
+            url: route.baseURL.appendingPathComponent(route.path),
+            cachePolicy: .reloadIgnoringLocalAndRemoteCacheData,
+            timeoutInterval: defaultTimeoutIntervalInSeconds
+        )
+
+        switch route.task {
+        case .request:
+            break
+        case let .requestWithURLParameters(urlParameters):
+            URLParametersEncoder.encode(urlParameters, into: &request)
+        case let .requestWithBodyParameters(bodyParameters):
+            do {
+                try JSONBodyEncoder.encode(bodyParameters, into: &request)
+            } catch {
+                print("Error encoding JSON body: \(error)")
+            }
+        }
```

Comment:
I suggest to make function `request(from:)` throwing and simplify this block. We don't handle errors in this place, so I don't see a need to use the `do-catch` block.
```suggestion
                try JSONBodyEncoder.encode(bodyParameters, into: &request)
        }
```

Reply:
I agree. Fixed with 203e327